code/modules/mob/living/carbon/human/species/station/station_special_abilities_vr.dm

```diff
@@ -209,188 +158,189 @@
-							if(client) client.images -= halbody
-							halbody = null
-			if(61 to 85) //25% chance
-				//food
-				if(!halbody)
-					var/list/possible_points = list()
-					for(var/turf/simulated/floor/F in view(src,world.view))
-						possible_points += F
-					if(possible_points.len)
-						var/turf/simulated/floor/target = pick(possible_points)
-						switch(rand(1,10))
-							if(1)
-								halbody = image('icons/mob/animal.dmi',target,"cow",TURF_LAYER)
-							if(2)
-								halbody = image('icons/mob/animal.dmi',target,"chicken",TURF_LAYER)
-							if(3)
-								halbody = image('icons/obj/food.dmi',target,"bigbiteburger",TURF_LAYER)
-							if(4)
-								halbody = image('icons/obj/food.dmi',target,"meatbreadslice",TURF_LAYER)
-							if(5)
-								halbody = image('icons/obj/food.dmi',target,"sausage",TURF_LAYER)
-							if(6)
-								halbody = image('icons/obj/food.dmi',target,"bearmeat",TURF_LAYER)
-							if(7)
-								halbody = image('icons/obj/food.dmi',target,"fishfillet",TURF_LAYER)
-							if(8)
-								halbody = image('icons/obj/food.dmi',target,"meat",TURF_LAYER)
-							if(9)
-								halbody = image('icons/obj/food.dmi',target,"meatstake",TURF_LAYER)
-							if(10)
-								halbody = image('icons/obj/food.dmi',target,"monkeysdelight",TURF_LAYER)
-
-						if(client) client.images += halbody
-						spawn(rand(50,80)) //Only seen for a brief moment.
-							if(client) client.images -= halbody
-							halbody = null
-			if(86 to 100) //15% chance
-				//disorientation
-				if(client)
-					client.dir = pick(2,4,8)
-					spawn(rand(20,50))
-						client.dir = 1
+							spawn(rand(10,30))
+								src << 'sound/weapons/Taser.ogg'
+					//Rare audio
+						if(12)
+	//These sounds are (mostly) taken from Hidden: Source
+							var/list/creepyasssounds = list('sound/effects/ghost.ogg', 'sound/effects/ghost2.ogg', 'sound/effects/Heart Beat.ogg', 'sound/effects/screech.ogg',\
+								'sound/hallucinations/behind_you1.ogg', 'sound/hallucinations/behind_you2.ogg', 'sound/hallucinations/far_noise.ogg', 'sound/hallucinations/growl1.ogg', 'sound/hallucinations/growl2.ogg',\
+								'sound/hallucinations/growl3.ogg', 'sound/hallucinations/im_here1.ogg', 'sound/hallucinations/im_here2.ogg', 'sound/hallucinations/i_see_you1.ogg', 'sound/hallucinations/i_see_you2.ogg',\
+								'sound/hallucinations/look_up1.ogg', 'sound/hallucinations/look_up2.ogg', 'sound/hallucinations/over_here1.ogg', 'sound/hallucinations/over_here2.ogg', 'sound/hallucinations/over_here3.ogg',\
+								'sound/hallucinations/turn_around1.ogg', 'sound/hallucinations/turn_around2.ogg', 'sound/hallucinations/veryfar_noise.ogg', 'sound/hallucinations/wail.ogg')
+							src << pick(creepyasssounds)
+				if(56 to 60) //5% chance
+					//Flashes of danger
+					//src << "Danger Flash"
+					if(!halbody)
+						var/list/possible_points = list()
+						for(var/turf/simulated/floor/F in view(src,world.view))
+							possible_points += F
```

Comment:
This is scary. It'd be a lot cheaper to just pick an x/y randomly within world.view until you found what you want than processing a whole full-screen view most of the time (making sure to bail if you fail too many times (or world.view*world.view times, since that'd be the whole area)).

Reply:
Hallucination code.